File: redis/src/main/java/info/xiaomo/redis/service/CityService.java
package info.xiaomo.redis.service;

import info.xiaomo.redis.model.CityInfo;
import org.springframework.cache.annotation.CacheConfig;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.stereotype.Component;

/**
 * @author : xiaomo
 */
@Component
@CacheConfig(cacheNames = "CityService")
public class CityService {

    @Cacheable
    public CityInfo getCity(int id, String city) {
        return new CityInfo(id, city);
    }
}

File: redis/src/main/java/info/xiaomo/redis/service/CityService.kt
package info.xiaomo.redis.service

import info.xiaomo.redis.model.CityInfo
import org.springframework.cache.annotation.CacheConfig
import org.springframework.cache.annotation.Cacheable
import org.springframework.stereotype.Component

/**
 * @author : xiaomo
 */
@Component
@CacheConfig(cacheNames = arrayOf("CityService"))
class CityService {

    @Cacheable
    fun getCity(id: Int, city: String): CityInfo {
        return CityInfo(id, city)
    }
}
